extensions/shared/notification-on-plan-upgrade.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * External dependencies
+ */
+import { dispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Shows a notification when a plan is marked as purchased
+ * after redirection from WPCOM.
+ */
+
+if ( window.location.search && -1 !== window.location.search.indexOf( 'plan' ) ) {
```

Comment:
Is this a common pattern that we adopt (naked logic in files that's executed on import)? I have not seen this pattern in a while and wonder if any ramifications with not exporting normally and calling from consumer instead.

Reply:
We generally like to avoid that, but we break that pattern sometimes, in particular in a Gutenberg context (due to the 'global' nature of e.g. registering blocks, or adding filters) -- see some of the other imports in [`extensions/editor.js`](https://github.com/Automattic/jetpack/pull/13209/files#diff-870b4d995e7728296254cd94b18a75e6). So this might be okay, at least for starters.

However, since I've worked a bit in server-side rendering, I'd like to guard against `window` use in contexts where it's not available, i.e. include a `typeof window !== undefined` check.

And finally, let's tighten the `window.location.search.indexOf( 'plan' )` check a bit more. Right now, it'll give false positives for a search string that contains any string containing `plan` -- eg `?edit=true&post=123&plantbased=yum`. This is obviously a rather pathological edge case, but it's easy enough to tighten. Let's use the `url` library to that end, see e.g. https://github.com/Automattic/jetpack/blob/be55a2ff9cd0e811741f34d3da2ca91565d4bf02/extensions/blocks/tiled-gallery/utils/index.js#L139

Reply:
That's perfect - will update.

> We generally like to avoid that, but we break that pattern sometimes, in particular in a Gutenberg context (due to the 'global' nature of e.g. registering blocks, or adding filters) -- see some of the other imports in extensions/editor.js. So this might be okay, at least for starters.

Yup, seen the other imports. Just not sure if it was necessary to be loading like this in `extensions/editor.js` (?).

Reply:
> Yup, seen the other imports. Just not sure if it was necessary to be loading like this in `extensions/editor.js` (?).

Yeah, good question. It'd be nice to colocate it more closely to the upgrade nudge, but then again, we only need the notice when we're not actually showing the nudge :sweat_smile: `register-jetpack-block.js` would then be another candidate, but it doesn't really seem like a better alternative to `editor.js`.